crates/cubecl-core/src/ir/operation.rs

```diff
@@ -1,50 +1,108 @@
 use std::fmt::Display;
 
-use super::{Branch, CoopMma, Subcube, Synchronization, Variable};
+use crate::prelude::AtomicOp;
+
+use super::{Branch, CoopMma, Item, Select, Subcube, Synchronization, Variable};
 use serde::{Deserialize, Serialize};
 
 /// All operations that can be used in a GPU compute shader.
 ///
 /// Notes:
 ///
-/// [Operator] and [Procedure] can be vectorized, but [Metadata] and [Branch] can't.
-/// Therefore, during tracing, only operators and procedures can be registered.
+/// [Operator] can be vectorized, but other operations can't.
+/// Therefore, during tracing, only operators can be registered.
 ///
-/// [Procedure] expansions can safely use all operation variants.
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
 #[allow(dead_code, missing_docs, clippy::large_enum_variant)] // Some variants might not be used with different flags
 pub enum Operation {
+    Assign(Variable),
```

Comment:
Normally it would be called copy but that was already used. Maybe we should just rename `Copy` to `CopyMemory` (which is in line with the SPIR-V name and `memcpy`), and call this one `Copy`. Because that's also how it's still used: to make a copy of a mutable variable.